src/launch/launcher.c

```diff
@@ -1359,22 +1366,35 @@ int launcher_run(Launcher *launcher) {
                 if (r)
                         return error_fold(r);
 
-                if (enabled && !supported)
-                        fprintf(stderr, "Kernel is missing AppArmor DBus support. Ignoring.\n");
-                else if (enabled)
-                        fprintf(stderr, "AppArmor enabled, but not supported. Ignoring.\n");
+                if (enabled && !supported) {
+                        fprintf(stderr, "Kernel is missing AppArmor DBus support.\n");
+                        policy.apparmor_mode = CONFIG_APPARMOR_DISABLED;
+                } else if (enabled) {
+                        policy.apparmor_mode = CONFIG_APPARMOR_ENABLED;
+                } else {
+                        policy.apparmor_mode = CONFIG_APPARMOR_DISABLED;
+                }
 
-                /* XXX: once the broker supports AppArmor, set this to DISABLED if and only if
-                 *      it is disabled in the kernel. */
-                policy.apparmor_mode = CONFIG_APPARMOR_DISABLED;
                 break;
         }
-        case CONFIG_APPARMOR_REQUIRED:
-                fprintf(stderr, "AppArmor required, but not supported. Exiting.\n");
+        case CONFIG_APPARMOR_REQUIRED: {
+                bool enabled, supported;
 
-                /* XXX: once the broker supports AppArmor, set this to enabled if and only
-                 *      if it is enabled in the kernel, and exit the launcher otherwise. */
-                return 0;
+                r = bus_apparmor_is_enabled(&enabled);
+                if (r)
+                        return error_fold(r);
+
+                r = bus_apparmor_dbus_supported(&supported);
+                if (r)
+                        return error_fold(r);
+
+                if (!enabled || !supported) {
+                        fprintf(stderr, "AppArmor required, but not supported. Exiting.\n");
+                        return 0;
+                }
+
+                policy.apparmor_mode = CONFIG_APPARMOR_ENABLED;
```

Comment:
maybe turn this entire thing into a helper function? We duplicate it now in initial-load and reload.